### ISSUE TITLE
No way to disable AVX

### ISSUE THREAD

**Issue by jengelh** [openucx/ucx#1548]
./configure provides no way to disable AVX. It would appear that if the compiler supports `-mavx`, openucx always activates -mavx, even if the machine itself cannot possibly run such code.

**Comment by yosefe** [comment]
@jengelh pls try `--without-avx`
```
$ ./configure --help|grep avx
  --with-avx              Use AVX compiler option.
```

**Comment by shamisp** [comment]
@yosefe - can we add C test before enabling this ?

**Comment by yosefe** [comment]
we already have it in configure stage

**Comment by shamisp** [comment]
I guess it can be a problem for distros that have to distribute x86 portable code. Probably we can add distro configure line option so it generates portable code.

**Comment by shamisp** [comment]
@yosefe sounds reasonable. This is something that has to be fixed for v1.2, otherwise distress will struggle to package this.
